aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -419,7 +445,10 @@ Tensor sparse_compressed_to_dense(
     dense = dense.reshape({n_batch, -1, values.size(-2), values.size(-1)});
     for (auto batch : c10::irange(n_batch)) {
       Tensor batch_indices = at::_convert_indices_from_csr_to_coo(
-          crow_indices[batch], col_indices[batch], false, false);
+          compressed_indices[batch],
+          plain_indices[batch],
+          false,
+          self.layout() == kSparseBsc);
       auto batch_row_indices = batch_indices.select(0, 0);
       auto batch_col_indices = batch_indices.select(0, 1);
```

Comment:
A follow-up on the point above: if batches are arbitrary strided, we could either reuse and/or write a new `batch_iterator_with_broadcasting`.

Reply:
Similar to the approach on the dense->compressed side. We can likely avoid `n_batch` kernel launches here. The reverse of the de-batching kernel used there could be used to generate `coo_indices` and create `dense` for a matrix with the batch dim folded into the compressed dim, and a reshape on that result restoring the batch dims. 

This work would be a good follow up if performance of this pathway needs improvement. 

Grated, that may not be *the* best way to do it, it should be *better*, than this.